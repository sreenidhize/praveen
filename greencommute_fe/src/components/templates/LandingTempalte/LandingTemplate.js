import React from "react";
import { Grid, makeStyles } from "@material-ui/core";
import PropTypes from "prop-types";

const useStyles = makeStyles((theme) => ({
  aqiClass: {
    backgroundColor: " #e7fce0",
    //alignItems: "center",
    //border: "1px solid blue",
  },
  gridClass: {
    height: "100vh",
    margin: "0",
  },
}));

function LandingTemplate(props) {
  const { leftContainer, rightContainer } = props;

  const classes = useStyles();
  return (
    <Grid container className={classes.gridClass}>
      <Grid item xs={7}>
        {leftContainer}
      </Grid>
      <Grid
        container
        item
        xs={5}
        className={classes.aqiClass}
        alignItems="center"
      >
        {rightContainer}
      </Grid>
    </Grid>
  );
}

LandingTemplate.propTypes = {
  leftContainer: PropTypes.any.isRequired,
  rightContainer: PropTypes.element.isRequired,
};

LandingTemplate.defultProps = {};

export default LandingTemplate;
