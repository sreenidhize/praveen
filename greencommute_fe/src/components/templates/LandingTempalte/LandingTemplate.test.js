import React from "react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import LandingTemplate from "./LandingTemplate";

describe("Template renders correctly", () => {
  test("to match snapshot", () => {
    const { asFragment, container } = render(
      <LandingTemplate
        leftContainer={<h1>Hello</h1>}
        rightContainer={<h2>Right</h2>}
      />
    );

    expect(asFragment()).toMatchSnapshot();
    expect(container.getElementsByTagName("h1").length).toBe(1);
    expect(container.getElementsByTagName("h2").length).toBe(1);
  });
});
