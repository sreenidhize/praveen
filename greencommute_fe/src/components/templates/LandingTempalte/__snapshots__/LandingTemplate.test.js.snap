// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Template renders correctly to match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root makeStyles-gridClass-2 MuiGrid-container"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7"
    >
      <h1>
        Hello
      </h1>
    </div>
    <div
      class="MuiGrid-root makeStyles-aqiClass-1 MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-grid-xs-5"
    >
      <h2>
        Right
      </h2>
    </div>
  </div>
</DocumentFragment>
`;
